circe/src/test/scala/org/http4s/circe/CirceSuite.scala

```diff
@@ -140,6 +141,26 @@ class CirceSuite extends JawnDecodeSupportSuite[Json] {
     writeToString(jsons).assertEquals("""[{"test1":"CirceSupport"},{"test2":"CirceSupport"}]""")
   }
 
+  test("stream json array decoder should decode JSON") {
```

Comment:
This is good.

What are the semantics if we have a stream that's ill-formed JSON?  I think we would return a successful stream that fails?  Otherwise, we're back to buffering to see if it's going to fail before we start streaming.  It would be neat to make this behavior clear with a test, though.

Reply:
> What are the semantics if we have a stream that's ill-formed JSON?

I... didn't think about that for some reason. I'll check and add a test case for it. Thanks!

Reply:
Currently, the IO blows up like this on malformed json:
```
==> X org.http4s.circe.test.CirceSuite.stream json array decoder should fail on malformed JSON  0.028s org.typelevel.jawn.ParseException: expected json value got 'C...' (line 1, column 36)
// stack trace
```

Reply:
By "successful stream that fails," I mean you get a `DecodeResult` that's a success that's a stream.  But compiling that stream will raise the error.  That's what's happening?

Reply:
> you get a DecodeResult that's a success that's a stream. But compiling that stream will raise the error

I pondered upon this for a while. That seems to be the case, yes. I added a few test cases, let me know if I got it right.